languages/cpp/reference/exercise-concepts/leap.md

```diff
@@ -0,0 +1,80 @@
+# Leap
+
+## Task
+
+Given a year, report if it is a leap year.
+
+
+## Example
+
+`leap.h`
+```cpp
+#ifndef LEAP_H
+#define LEAP_H
+
+namespace leap
+{
+    bool is_leap_year(int year);
+}
+
+#endif
+```
+
+`leap.cpp`
+```cpp
+#include "leap.h"
+
+bool leap::is_leap_year(int year)
+{
+    return year % 4 == 0 && (year % 100 != 0 || year % 400 == 0);
+}
+```
+
+
+## Concepts
+
+- .h file: the most common approach is to use a `.h` for the declarations
```

Comment:
We gotta discuss whether we promote .cpp/.hpp pairs or not

Reply:
Personally I make the strict separation between .c/.h and .cpp/.hpp, rationale being that it allows me to identify the language of a given header in shared langs repos.

Reply:
Exercism [promotes ](https://teams.exercism.io/values) "fluency". IMHO that means a student should be able to handle both. But I think .h/.cpp files separate interface and implementation and its better to learn them first.

Reply:
From the [C++ Core Guidelines](https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Rs-file-suffix): "SF.1: Use a .cpp suffix for code files and .h for interface files if your project doesn’t already follow another convention".